spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -604,6 +604,48 @@
         expect(subject.send(:event_action_tuple, event)[1]).not_to include(:pipeline)
       end
     end
+
+    context "with both pipeline and target_ingest_pipeline" do
+      let(:options) { {"pipeline" => "%{pipeline}" } }
+      let(:event) { LogStash::Event.new({"pipeline" => "my-ingest-pipeline", "[@metadata][target_ingest_pipeline]" => "meta-ingest-pipeline"}) }
+
+      it "should interpolate the pipeline value and set it" do
+        expect(subject.send(:event_action_tuple, event)[1]).to include(:pipeline => "my-ingest-pipeline")
+      end
```

Comment:
nit: we certainly have a mix, but since this is a spec, using _declarative_ language about what this plugin _does_ is more clear than using language that says what it _should do_ ([more in the ruby style guide](https://github.com/rubocop/rspec-style-guide#should-in-example-docstrings)).

also, we should catch the case where the plugin's `pipeline` is specified but it doesn't get sprintf expanded, something like:

```suggestion
      it "interpolates the plugin's pipeline value" do
        expect(subject.send(:event_action_tuple, event)[1]).to include(:pipeline => "my-ingest-pipeline")
      end
      context "when the plugin's `pipeline` is constant" do
        let(:options) { super().merge("pipeline" => "my-constant-pipeline") }
         it "uses plugin's pipeline value" do
          expect(subject.send(:event_action_tuple, event)[1]).to include(:pipeline => "my-constant-pipeline")
        end
      end
      context "when the plugin's `pipeline` includes an unresolvable sprintf placeholder" do
        let(:options) { super().merge("pipeline" => "reference-%{unset}-field") }
        it "does not use the target_ingest_pipeline" do
          # when sprintf doesn't resolve a placeholder, the behaviour of our `pipeline` is UNSPECIFIED.
          # here we only validate that the presence of the magic field does not
          # override an explicitly-configured pipeline.
          expect(subject.send(:event_action_tuple, event)[1]).to_not include(:pipeline => "my-ingest-pipeline")
        end
      end
```